### PR TITLE
[py-tx] CONTRIBUTING.MD updates

### PR DESCRIPTION
**Pull request by Dcallies**
More information on expectations for testing, typing, and pull requests

### CODE REVIEW

File: python-threatexchange/CONTRIBUTING.md [hunk 1]
```diff
@@ -12,9 +13,36 @@ black ./python-threatexchange/
 Additionally, your IDE may have support for automatically re-formatting your source files
 using black through your IDE settings.
 
-### Running Tests
-python-threatexchange is a bit short on tests, but you could help fix that.
-To run the tests `make test`
+When in doubt, follow the style of the file you are editing, and if starting new files, be consistent within.
+
+## python typing and `mypy`
+We aiming for full strict coverage via PEP484 tying annotations. However, this is a work in progress. We use [mypy](https://mypy.readthedocs.io/en/stable/index.html) to verifying typing.
```
Comment (schatten) [inline python-threatexchange/CONTRIBUTING.md:19]: spell: typing → typing. verifying → verify.

File: python-threatexchange/CONTRIBUTING.md [hunk 2]
```diff
@@ -12,9 +13,36 @@ black ./python-threatexchange/
...
+
+### `# type: ignore`
+We're not all typing experts yet, and we've confused ourselves with Generics, Aliases and object heirarchies, and occasionally `# type: ignored`. If you end up in a sticky situation and decide to throw in the towel on typing and use `# type: ignore`, use `mypy --show-error-codes` to only ignore the specific error code, and expect discussion on the PR on the level of effort needed to make the code compliant.
+
+### New Files or Directories
+When creating new files or directories, please first try and make them compatability with `--strict` to do this, add in the mypy.ini file
+```
+[path.to.module]
```
Comment (schatten) [inline python-threatexchange/CONTRIBUTING.md:27]: Can we do the reverse? Add all current files with strict=False (one time) and expect all new files to be strict=True?

**Review by schatten — APPROVED** [review]
Looks good. Thanks for doing the docs. This is super useful (hopefully).
Some spelling and grammar corrections are highlighted inline.

I have a question on the strict=True / False workflow for mypy too. Let's not hold this PR up. Feel free to merge after spelling and grammar corrections.

File: python-threatexchange/CONTRIBUTING.md (hunk 2 above)
Comment (Dcallies) [reply python-threatexchange/CONTRIBUTING.md:27]: Basically all files fail strict right now :/ I tried adding the strict flags one at a time in #1086, but found we're hundreds of errors away, and there's not a super clean way to tag all the files. I figured it'd be easy to do one directory at a time.

File: python-threatexchange/CONTRIBUTING.md (hunk 1 above)
Comment (Dcallies) [reply python-threatexchange/CONTRIBUTING.md:19]: > typing → typing

😵

**Pull request merged**